lib/chef/node/mixin/immutablize_hash.rb

```diff
@@ -128,9 +128,37 @@ module ImmutablizeHash
           values
           values_at
           zip
+          deep_merge
```

Comment:
Mind splitting the Immutable changes into a separate PR? These mixins are really finicky performance-wise, and I wouldn't typically look for changes here in a merge about licensing code

Reply:
@dafyddcrosby  Thanks for the feedback but `chef-licensing` gem is pulling in `activesupport` as a dependency which has given rise to introducing bunch of these actions via rails, hence these fixes. But yes we understand how picky these mixins are in behaviour, and we are being careful before these fixes are merged.

Reply:
These changes can be done independently from the rest of the licensing code changes FWICT, and we can backport it to the Chef 18 branch, which means any perf implications can be isolated to that change in immutablize_hash. By backporting it to Chef 18, we can work out any perf kinks that otherwise have nothing to do with the licensing code. It's not a matter of 'being careful' as much as being able to tell if a perf problem is due to that particular change in immutablize_hash. Splitting it into a separate PR+commit reduces risk migrating from 18 to 19.

Reply:
Just fyi , licensing will not be backported to chef 18 cc:  @tpowell-progress @neha-p6

Reply:
@dafyddcrosby, would it be okay if I create a new PR against the branch of this one, containing only the immutable changes? This way, the changes would eventually merge into this PR, but they’d also have a dedicated PR for visibility. Would that work?

Reply:
@ahasunos  That would be perfect 👍 
@rahulgoel1 I know licensing won't be backported to 18, my concern here is only the immutable changes. The immutable changes could be backported to Chef 18 FWICT, and then it's one less thing that makes the 18 branch different from 19. Apologies if that's what you meant, just want to be clear 😄

Reply:
Raised #14667 @dafyddcrosby